[dynamo, 3.12] xfail refleaking tests due to buggy getattr_static

williamwen42 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #125062

For tracking https://github.com/pytorch/pytorch/issues/124302 so that we can re-enable the test once 3.12 updates with the bug fix for https://github.com/python/cpython/issues/118013.

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @chenyang78 @kadeng @chauhang @amjames

williamwen42 committed:
[dynamo, 3.12] xfail refleaking tests due to buggy getattr_static

[ghstack-poisoned]

williamwen42 committed:
Update on "[dynamo, 3.12] xfail refleaking tests due to buggy getattr_static"


For tracking https://github.com/pytorch/pytorch/issues/124302 so that we can re-enable the test once 3.12 updates with the bug fix for https://github.com/python/cpython/issues/118013.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang

[ghstack-poisoned]

williamwen42 committed:
Update test_misc.py

anijain2305 commented on test/dynamo/test_misc.py:
```diff
@@ -10167,14 +10172,14 @@ def __init__(self):
+                    return self.fc_ref(x[0])
 
             # return fc to keep it alive in _test_compile_model_free
             return Mod(), (torch.randn(100, 100), fc)
```
Can you check if returning 'fc' here works correctly for '_test_compile_model_free'. The call to 'model_input_ptr' returns a tuple of 3 elements. But the code expects mod and inp. It works somehow without failing today, but causes failure for my stack.

williamwen42 replied:
In 3.12 the layer leaks due to `getattr_static` if we call `self.fc_ref` but not `fc`. I think I originally intended to use `self.fc_ref` (otherwise, why would I define it?).

The return of `model_inp_ctr` is a little tricky but it returns 2 things - the second returned item is a tuple of input and layer, but the forward function only uses the input.

williamwen42 committed:
Update

[ghstack-poisoned]

williamwen42 commented:
@pytorchbot merge

williamwen42 committed:
Update on "[dynamo, 3.12] xfail refleaking tests due to buggy getattr_static"


For tracking https://github.com/pytorch/pytorch/issues/124302 so that we can re-enable the test once 3.12 updates with the bug fix for https://github.com/python/cpython/issues/118013.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang

[ghstack-poisoned]

williamwen42 committed:
Update

[ghstack-poisoned]

williamwen42 committed:
Update on "[dynamo, 3.12] xfail refleaking tests due to buggy getattr_static"


For tracking https://github.com/pytorch/pytorch/issues/124302 so that we can re-enable the test once 3.12 updates with the bug fix for https://github.com/python/cpython/issues/118013.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang

[ghstack-poisoned]

williamwen42 commented:
@pytorchbot merge
